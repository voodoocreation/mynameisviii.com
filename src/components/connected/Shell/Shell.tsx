import cn from "classnames";
import Head from "next/head";
import * as React from "react";
import { FormattedMessage, InjectedIntl, injectIntl } from "react-intl";
import { connect } from "react-redux";

import { isServer } from "../../../helpers/dom";
import Banner from "../../presentation/Banner/Banner";
import Button from "../../presentation/Button/Button";
import Loader from "../../presentation/Loader/Loader";
import OnlineStatusToast from "../../presentation/OnlineStatusToast/OnlineStatusToast";
import Toast from "../../presentation/Toast/Toast";
import ToastContainer from "../../presentation/ToastContainer/ToastContainer";

import * as selectors from "../../../selectors/root.selectors";

interface IStoreProps {
  hasNewVersion: boolean;
  isLoading: boolean;
  isOnline: boolean;
}

interface IProps extends IStoreProps {
  className?: string;
  intl: InjectedIntl;
}

class Shell extends React.Component<IProps> {
  public componentWillMount() {
    if (!isServer()) {
      const html = document.documentElement as HTMLHtmlElement;
      html.classList.add("isClientRendered");
    }
  }

  public render() {
    const {
      children,
      className,
      hasNewVersion,
      isLoading,
      isOnline
    } = this.props;
    const { formatMessage } = this.props.intl;

    return (
      <article className={cn("Page", { isLoading }, className)}>
        <Head>
          <meta
            property="og:site_name"
            content={formatMessage({ id: "BRAND_NAME" })}
          />
        </Head>

        <Banner />

        <main className="Page-body" role="main">
          {isLoading ? <Loader className="PageLoader" /> : children}
        </main>

        <ToastContainer>
          <OnlineStatusToast isOnline={isOnline} />

          <Toast
            className="HasNewVersionToast"
            hasAutoDismiss={false}
            isVisible={hasNewVersion}
          >
            <FormattedMessage id="NEW_VERSION_AVAILABLE" />
            <Button
              className="HasNewVersionToast-refreshButton"
              onClick={this.onRefreshClick}
            >
              <FormattedMessage id="REFRESH" />
            </Button>
          </Toast>
        </ToastContainer>
      </article>
    );
  }

  private onRefreshClick = () => {
    window.location.reload(true);
  };
}

const mapStateToProps = (state: any) => ({
  hasNewVersion: selectors.hasNewVersion(state),
  isLoading: selectors.getPageIsLoading(state),
  isOnline: selectors.isOnline(state)
});

export default injectIntl<any>(connect<IStoreProps>(mapStateToProps)(Shell));
