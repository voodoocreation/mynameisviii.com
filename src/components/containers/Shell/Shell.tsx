import cn from "classnames";
import Head from "next/head";
import * as React from "react";
import { InjectedIntl, injectIntl } from "react-intl";
import { connect } from "react-redux";

import Banner from "../../presentation/Banner/Banner";

import * as selectors from "../../../selectors/root.selectors";
import Loader from "../../presentation/Loader/Loader";

interface IStoreProps {
  isLoading: boolean;
}

interface IProps extends IStoreProps {
  className?: string;
  intl: InjectedIntl;
}

class Shell extends React.Component<IProps> {
  public componentWillMount() {
    if (
      typeof document !== "undefined" &&
      !document.documentElement.classList.contains("isClientRendered")
    ) {
      document.documentElement.classList.add("isClientRendered");
    }
  }

  public render() {
    const { children, className, isLoading } = this.props;
    const { formatMessage } = this.props.intl;

    return (
      <article className={cn("Page", { isLoading }, className)}>
        <Head>
          <meta
            property="og:site_name"
            content={formatMessage({ id: "BRAND_NAME" })}
          />
        </Head>

        <Banner />

        <main className="Page-body" role="main">
          {isLoading ? <Loader className="PageLoader" /> : children}
        </main>
      </article>
    );
  }
}

const mapStateToProps = (state: any) => ({
  isLoading: selectors.getPageIsLoading(state)
});

export default injectIntl<any>(connect<IStoreProps>(mapStateToProps)(Shell));
