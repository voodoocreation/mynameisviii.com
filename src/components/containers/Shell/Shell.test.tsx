import { render } from "enzyme";
import * as React from "react";

import Shell from "./Shell";

jest.mock("react-redux", () => ({
  connect: () => (component: any) => component
}));

const setup = (fn: any, fromTestProps?: any) => {
  const props = {
    className: "TestClassName",
    ...fromTestProps
  };

  return {
    actual: fn(<Shell {...props} />),
    props
  };
};

describe("<Shell />", () => {
  it("renders correctly", () => {
    const { actual } = setup(render);
    expect(actual).toMatchSnapshot();
  });

  it("renders correctly when loading", () => {
    const { actual } = setup(render, {
      isLoading: true
    });
    expect(actual).toMatchSnapshot();
  });
});
