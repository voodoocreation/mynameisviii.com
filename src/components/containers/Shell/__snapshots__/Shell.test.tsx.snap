// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[containers] <Shell /> renders correctly 1`] = `
<article
  class="Page TestClassName"
>
  <header
    class="Banner"
    role="banner"
  >
    <a
      class="Brand"
      href="/"
    >
      <span>
        My name is Viii: a painter of frequencies and thoughts
      </span>
      <svg />
      <svg />
    </a>
    <nav
      class="Navigation"
      role="navigation"
    >
      <button>
        <span>
          Menu
        </span>
      </button>
      <ul>
        <li>
          <a
            href="/news"
          >
            <span>
              News
            </span>
          </a>
        </li>
        <li>
          <a
            href="/releases"
          >
            <span>
              Releases
            </span>
          </a>
        </li>
        <li>
          <a
            href="/stems"
          >
            <span>
              Stems
            </span>
          </a>
        </li>
        <li>
          <a
            href="/appearances"
          >
            <span>
              Appearances
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="Page-body"
    role="main"
  />
</article>
`;

exports[`[containers] <Shell /> renders correctly when loading 1`] = `
<article
  class="Page isLoading TestClassName"
>
  <header
    class="Banner"
    role="banner"
  >
    <a
      class="Brand"
      href="/"
    >
      <span>
        My name is Viii: a painter of frequencies and thoughts
      </span>
      <svg />
      <svg />
    </a>
    <nav
      class="Navigation"
      role="navigation"
    >
      <button>
        <span>
          Menu
        </span>
      </button>
      <ul>
        <li>
          <a
            href="/news"
          >
            <span>
              News
            </span>
          </a>
        </li>
        <li>
          <a
            href="/releases"
          >
            <span>
              Releases
            </span>
          </a>
        </li>
        <li>
          <a
            href="/stems"
          >
            <span>
              Stems
            </span>
          </a>
        </li>
        <li>
          <a
            href="/appearances"
          >
            <span>
              Appearances
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="Page-body"
    role="main"
  >
    <span
      class="Loader"
    >
      <div
        class="Loader-animation"
      >
        <span />
        <span />
        <span />
      </div>
    </span>
  </main>
</article>
`;
